test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs

```diff
@@ -202,6 +202,63 @@ public void AutoFitRecommendationTest()
             Assert.NotEqual(0, metrices.MeanSquaredError);
         }
 
+        [Fact]
+        public void AutoFitWithPresplittedData()
+        {
+            // Models created in AutoML should work over the same data,
+            // no matter how that data is splitted before passing it to the experiment execution
+            // or to the model for prediction
```

Comment:
This test abstracts the issue in #5256 (which is also the issue in #4048 and https://github.com/dotnet/machinelearning-modelbuilder/issues/694 )

This test fails with a 
`System.ArgumentOutOfRangeException : Could not find input column 'SamplingKeyColumn'` without the changes on this PR.

The particular change that fixed this issue is dropping the (temporary version of) `SamplingKeyColumn`, on `DataOperationsCatalog.CrossValidationSplit`.